src/Database/Driver.php

```diff
@@ -523,6 +523,16 @@ public function getConnectRetries(): int
         return $this->connectRetries;
     }
 
+    /**
+     * Returns the connection role this driver performs.
+     *
+     * @return string
+     */
+    public function getRole(): string
+    {
+        return $this->_config['_role'] ?? Connection::ROLE_WRITE;
+    }
```

Comment:
Should this be getConnectionRole to match Query?

Reply:
I don't think it needs to match. Query's benefit from the additional scoping as `getRole()` by itself could be related to application logic roles or something else. I don't think we have that ambiguity here.